src/services/db/GitFileSystemService.ts

```diff
@@ -1066,37 +1136,26 @@ export default class GitFileSystemService {
     repoName: string,
     branchName: string
   ): ResultAsync<GitHubCommitData, GitFileSystemError> {
-    return ResultAsync.fromPromise(
-      this.git.cwd(`${EFS_VOL_PATH}/${repoName}`).log([branchName]),
-      (error) => {
-        logger.error(`Error when getting latest commit of branch: ${error}`)
-
-        if (error instanceof GitError) {
-          return new GitFileSystemError(
-            "Unable to retrieve branch log info from disk"
-          )
+    return this.getGitLog(repoName, branchName)
+      .orElse(() => this.getGitLog(repoName, `origin/${branchName}`))
+      .andThen((logSummary) => {
```

Comment:
nit: possible to add a type for `logSummary`?

Reply:
Hmm the type for `logSummary` is declared in the return type of `this.getGitLog()`